### PR TITLE
Add better mw chaining

### PR DESCRIPTION
**Pull request by pschwartz**

### CODE REVIEW

**Comment by pschwartz** [comment]
Don't merge this!

File: web/middleware/auth.go
Comment (rattboi) [inline web/middleware/auth.go]: What's the use-case for this? It isn't used yet?

File: web/engine/handlers.go
Comment (rattboi) [inline web/engine/handlers.go:31]: I'm annoyed that so much of this needs hand-rolled. My lack of GoLang background means I don't know a better option, but it feels like a lot of re-inventing the wheel is happening here.

**Review by rattboi — COMMENTED** [review]
It looks fine. Hope we can get tests in place soon, as there's a lot here to test. I'll change this to approve once there are responses to my comments.

File: web/engine/handlers.go
Comment (pschwartz) [reply web/engine/handlers.go:31]: And here it is not lack of golang background. The ability is not there. #go-nuts on freenode suggested doing a loop as the only other way to do it in the language is with the sort package and implementing  a sorter for the type which is more work then a single function.

File: web/middleware/auth.go
Comment (pschwartz) [reply web/middleware/auth.go]: I will remove this. I didn't realize I left it. I moved the api endpoints for modifying a status to use the authenticated user instead of passing a user in as we only want a user to modify their own.

**Review by rattboi — APPROVED** [review]

**Pull request merged**
